chimg/chroot.py

```diff
@@ -215,6 +251,31 @@ def _snap_install(self, name: str, channel: str, classic: bool = False, revision
                 name=name, filename=os.path.basename(snap_file), channel=channel, classic=classic, info=snap_info_yaml
             )
 
+    def _snaps_already_installed(self) -> Dict[str, SnapInfo]:
+        """
+        Collect information about already preseeded snaps from /var/lib/snapd/seed/seed.yaml
+        """
+        snap_infos: Dict[str, SnapInfo] = {}
+
+        seed_yaml = self._snaps_read_seed_yaml()
+        if not seed_yaml:
+            logger.info("no seed.yaml file exist so no snaps already preseeded.")
+            return snap_infos
+
+        for snap in seed_yaml["snaps"]:
+            # path to the .snap file
+            p = Path(f"{self._ctx.chroot_path}/{SNAPD_SEED_DIR}/snaps/{snap['file']}")
+            snap_info_yaml = self._snap_info(p.as_posix())
```

Comment:
TIL: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix

Cool if it works on on Windows too :-D